src/plugins/console/public/application/hooks/use_send_current_request_to_es/send_request_to_es.ts

```diff
@@ -107,37 +115,48 @@ export function sendRequestToES(args: EsRequestArgs): Promise<ESRequestResult[]>
             });
 
             // single request terminate via sendNextRequest as well
-            sendNextRequest();
+            await sendNextRequest();
+          }
+        }
+      } catch (error) {
+        let value;
+        let contentType = '';
+        if (!error) {
+          value =
+            "\n\nFailed to connect to Console's backend.\nPlease check the Kibana server is up and running";
+        }
+
+        if (error.response) {
+          const { status, headers } = error.response;
+
+          if (error.body) {
+            value = JSON.stringify(error.body, null, 2); // ES error should be shown
+            contentType = headers.get('Content-Type');
           } else {
-            let value;
-            let contentType: string;
-            if (xhr.responseText) {
-              value = xhr.responseText; // ES error should be shown
-              contentType = xhr.getResponseHeader('Content-Type');
-            } else {
-              value = 'Request failed to get to the server (status code: ' + xhr.status + ')';
-              contentType = 'text/plain';
-            }
-            if (isMultiRequest) {
-              value = '# ' + req.method + ' ' + req.url + '\n' + value;
-            }
-            reject({
-              response: {
-                value,
-                contentType,
-                timeMs: Date.now() - startTime,
-                statusCode: xhr.status,
-                statusText: xhr.statusText,
-              },
-              request: {
-                data: esData,
-                method: esMethod,
-                path: esPath,
-              },
-            });
+            value = 'Request failed to get to the server (status code: ' + status + ')';
+            contentType = headers.get('Content-Type');
+          }
+
+          if (isMultiRequest) {
+            value = '# ' + req.method + ' ' + req.url + '\n' + value;
           }
         }
-      );
+
+        reject({
+          response: {
+            value,
+            contentType,
+            timeMs: Date.now() - startTime,
+            statusCode: error?.response?.status ?? 0,
```

Comment:
nit: I think we should use one of the defined error status codes here instead of `0` as a default, maybe `500` would make more sense?